patches/bess/0009-Augment-IPChecksum-and-L4Checksum-modules-to-support.patch

```diff
@@ -0,0 +1,168 @@
+From 48ea01c14783b55aa680d1d51da42f81bd9c4c54 Mon Sep 17 00:00:00 2001
+From: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
+Date: Tue, 28 Apr 2020 14:53:18 -0700
+Subject: [PATCH] Augment IPChecksum and L4Checksum modules to support
+ verifications.
+
+Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>
+---
+ core/modules/ip_checksum.cc | 22 +++++++++++++++++-----
+ core/modules/ip_checksum.h  |  7 +++++++
+ core/modules/l4_checksum.cc | 31 +++++++++++++++++++++++--------
+ core/modules/l4_checksum.h  |  7 +++++++
+ 4 files changed, 54 insertions(+), 13 deletions(-)
+
+diff --git a/core/modules/ip_checksum.cc b/core/modules/ip_checksum.cc
+index ab7014f4..2dbb0a3b 100644
+--- a/core/modules/ip_checksum.cc
++++ b/core/modules/ip_checksum.cc
+@@ -34,6 +34,8 @@
+ #include "../utils/ether.h"
+ #include "../utils/ip.h"
+ 
++enum { DEFAULT_GATE = 0, FORWARD_GATE };
++
+ void IPChecksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
+   using bess::utils::be16_t;
+   using bess::utils::Ethernet;
+@@ -54,7 +56,8 @@ void IPChecksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
+       data = qinq + 1;
+       ether_type = qinq->ether_type;
+       if (ether_type != be16_t(Ethernet::Type::kVlan)) {
+-        continue;
++	EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
++	continue;
+       }
+     }
+ 
+@@ -67,13 +70,22 @@ void IPChecksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
+     if (ether_type == be16_t(Ethernet::Type::kIpv4)) {
+       ip = reinterpret_cast<Ipv4 *>(data);
+     } else {
++      EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
+       continue;
+     }
+ 
+-    ip->checksum = CalculateIpv4Checksum(*ip);
++    if (verify)
++      EmitPacket(ctx, batch->pkts()[i], (VerifyIpv4Checksum(*ip)) ? FORWARD_GATE : DEFAULT_GATE);
++    else {
++      ip->checksum = CalculateIpv4Checksum(*ip);
++      EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
```

Comment:
When there is no verify there should be no difference in behavior to today. Gate 0 should be used

Reply:
Guess it will mess up the continue logic